test-cypress/integration/repository/repositories.spec.js

```diff
@@ -0,0 +1,404 @@
+        getRepositoryFromList(secondRepoId)
+            .find('.icon-connection-on')
+            .should('be.visible');
+        getRepositoriesDropdown()
+            .find('.active-repository')
+            .should('contain', secondRepoId);
+        // The first should return back to the dropdown items
+        getRepositoriesDropdown()
+            .click()
+            .find('.dropdown-menu .dropdown-item')
+            .should('contain', repositoryId);
+        // Hide the menu
+        getRepositoriesDropdown().click();
+
+        // Make the first one default, clear the cookies and reload the page
+        // Note: the first one should be cleared in afterEach() preventing other tests to select it by default
+        getRepositoryFromList(repositoryId)
+            .find('.pin-repository-btn')
+            .should('not.have.class', 'active')
+            .click()
+            .should('have.class', 'active');
+        // The currently selected repository is kept as a cookie
+        cy.clearCookies();
+        cy.visit('/repository');
+
+        // Should automatically select the default repository
+        getRepositoriesDropdown()
+            .find('.active-repository')
+            .should('contain', repositoryId);
+
+        // TODO: Could push the repo in an array and afterEach can delete all of them
+        cy.deleteRepository(secondRepoId);
+    });
+
+    it('should allow to edit existing repository', () => {
+        const newTitle = 'Title edit';
+        const newBaseUrl = 'http://example.org/wine#';
+
+        createRepository();
+        typeRepositoryId(repositoryId);
+        typeRepositoryTitle('Title');
+        saveRepository();
+
+        editRepository(repositoryId);
+
+        // Some fields should be disabled
+        getRepositoryRulesetMenu().should('be.disabled');
+        getRepositoryDisableSameAsCheckbox().should('be.disabled');
+
+        typeRepositoryTitle(newTitle);
+        typeRepositoryBaseURL(newBaseUrl);
+        getRepositoryContextIndexCheckbox().check();
+
+        getSaveRepositoryButton().click();
+        confirmModal();
+        waitLoader();
+
+        // See the title is rendered in the repositories list
+        getRepositoryFromList(repositoryId).should('contain', newTitle);
+
+        // Go to edit again to verify changes
+        editRepository(repositoryId);
+
+        getRepositoryTitleField().should('have.value', newTitle);
+        getRepositoryBaseURLField().should('have.value', newBaseUrl);
+        getRepositoryContextIndexCheckbox().should('have.value', 'true');
+    });
+
+    it('should allow to delete existing repository', () => {
+        createRepository();
+        typeRepositoryId(repositoryId);
+        saveRepository();
+
+        selectRepoFromDropdown(repositoryId);
+
+        getRepositoryFromList(repositoryId)
+            .find('.repository-actions .delete-repository')
+            // Forcefully clicking it due to https://github.com/cypress-io/cypress/issues/695
```

Comment:
Have you tried this suggestion
https://github.com/cypress-io/cypress/issues/695#issuecomment-379817133

Reply:
Just did and it doesn't work